packages/tpetra/core/src/Tpetra_Details_getDiagCopyWithoutOffsets_def.hpp

```diff
@@ -130,8 +130,8 @@ class GetLocalDiagCopyWithoutOffsetsNotFillCompleteFunctor {
 
     // Side effects start below this point.
 
-    diag.template modify<Kokkos::HostSpace> ();
-    D_lcl_ = diag.template getLocalView<Kokkos::HostSpace> ();
+    diag.modify_host ();
+    D_lcl_ = diag.getDualView().view_host ();
```

Comment:
Why call `getDualView()` here?  Why not just call `getLocalViewHost()`?  We should deprecate and remove getDualView() from MultiVector, since it exposes an implementation detail of how Tpetra implements dual view semantics.

Reply:
@crtrott I think you made the change Mark is asking about here, any reason to not use `getLocalViewHost`?

Reply:
@mhoemmen fixed this case in the kokkos-promotion branch, there may be a few more to look out for on that branch.